apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationResult.cs

```diff
@@ -32,10 +32,16 @@ public sealed class TranslationResult
         public string TranslatedText { get; }
 
         /// <summary>
-        /// The source language code, either as provided by the caller
-        /// or detected by the server.
+        /// The source language code detected by the server. This may
+        /// be null, typically due to the source language being supplied
+        /// by the caller.
         /// </summary>
-        public string SourceLanguage { get; }
+        public string DetectedSourceLanguage { get; }
+
+        /// <summary>
+        /// The source language code supplied by the caller, if any.
+        /// </summary>
+        public string SpecifiedSourceLanguage { get; }
```

Comment:
Since there is a `Language` class, which itself has names and codes, these names could cause a bit of confusion. Perhaps suffixing each with `...Code` would be good, although I'm not sure it's worth the extra name length.